contracts/golos.worker/main.cpp

```diff
@@ -0,0 +1,1088 @@
+      proposal.set_state(proposal_t::STATE_WORK);
+    });
+  }
+
+  /**
+   * @brief cancelwork cancels work. Can be called by the worker or technical specification author
+   * @param proposal_id propsal ID
+   * @param initiator a cancel initiator's account name
+   */
+  /// @abi action
+  void cancelwork(proposal_id_t proposal_id, account_name initiator)
+  {
+    LOG("proposal_id: %, initiator: %", proposal_id, ACCOUNT_NAME_CSTR(initiator));
+    auto proposal_ptr = get_proposal(proposal_id);
+    eosio_assert(proposal_ptr->state == proposal_t::STATE_WORK, "invalid proposal state");
+    eosio_assert(proposal_ptr->type == proposal_t::TYPE_1, "unsupported action");
+
+    if (initiator == proposal_ptr->worker)
+    {
+      require_auth(proposal_ptr->worker);
+    }
+    else
+    {
+      require_auth(proposal_ptr->tspec_author);
+    }
+
+    get_proposals().modify(proposal_ptr, initiator, [&](proposal_t &proposal) {
+      refund(proposal, initiator);
+    });
+  }
+
+  /**
+   * @brief poststatus post status for the work done
+   * @param proposal_id proposal ID
+   * @param comment_id comment ID
+   * @param comment comment data
+   * @param finished true if all work done
+   */
+  /// @abi action
+  void poststatus(proposal_id_t proposal_id, comment_id_t comment_id, const comment_data_t &comment, bool finished)
+  {
+    LOG("proposal_id: %, comment: %, final: %", proposal_id, comment.text.c_str(), (int) finished);
+    auto proposal_ptr = get_proposal(proposal_id);
+    eosio_assert(proposal_ptr->state == proposal_t::STATE_WORK, "invalid proposal state");
+    eosio_assert(proposal_ptr->type == proposal_t::TYPE_1, "unsupported action");
+    require_auth(proposal_ptr->worker);
+
+    get_proposals().modify(proposal_ptr, proposal_ptr->worker, [&](auto &proposal) {
+      proposal.work_status.add(comment_id, proposal.worker, comment);
+
+      if (finished)
+      {
+        proposal.set_state(proposal_t::STATE_TSPEC_AUTHOR_REVIEW);
+      }
+    });
+  }
+
+  /**
+   * @brief acceptwork accepts a work that was done by the worker. Can be called only by the technical specification author
+   * @param proposal_id proposal ID
+   * @param comment_id comment ID
+   * @param comment
+   */
+  /// @abi action
+  void acceptwork(proposal_id_t proposal_id, comment_id_t comment_id, const comment_data_t &comment)
+  {
+    LOG("proposal_id: %, comment: %", proposal_id, comment.text.c_str());
+    auto proposal_ptr = get_proposal(proposal_id);
+    eosio_assert(proposal_ptr->state == proposal_t::STATE_TSPEC_AUTHOR_REVIEW, "invalid proposal state");
+    eosio_assert(proposal_ptr->type == proposal_t::TYPE_1, "unsupported action");
+    require_auth(proposal_ptr->tspec_author);
+
+    get_proposals().modify(proposal_ptr, proposal_ptr->tspec_author, [&](auto &proposal) {
+      proposal.set_state(proposal_t::STATE_DELEGATES_REVIEW);
+      proposal.work_status.add(comment_id, proposal.tspec_author, comment);
```

Comment:
this "comment" is different, not sure it fits in `work_status`. at least it must be marked somehow

Reply:
It was initially designed to be the last comment.